NeoMathEngine/include/NeoMathEngine/MathEngineDropout.h

```diff
@@ -0,0 +1,42 @@
+/* Copyright © 2017-2024 ABBYY
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <NeoMathEngine/NeoMathEngine.h>
+#include <NeoMathEngine/CrtAllocatedObject.h>
+
+namespace NeoML {
+
+struct CBaseDropoutDesc : public CDropoutDesc {
```

Comment:
Hide `CBaseDropoutDesc();` to private, this class should not be created itself

Reply:
and all fields :)

Reply:
why protect fields?( we need to access them outside, or i misunderstood something?

Reply:
See my new comments, we should not allow access